pkg/odo/cli/debug/info.go

```diff
@@ -59,9 +62,13 @@ func (o InfoOptions) Validate() error {
 // Run implements all the necessary functionality for port-forward cmd.
 func (o InfoOptions) Run() error {
 	if debugFileInfo, debugging := debug.GetDebugInfo(o.PortForwarder); debugging {
-		log.Infof("Debug is running for the component on the local port : %v\n", debugFileInfo.LocalPort)
+		if log.IsJSON() {
+			machineoutput.OutputSuccess(debugFileInfo)
+		} else {
+			log.Infof("Debug is running for the component on the local port : %v", debugFileInfo.Spec.LocalPort)
+		}
 	} else {
-		log.Infof("Debug is not running for the component %v\n", o.localConfigInfo.GetName())
+		return fmt.Errorf("debug is not running for the component %v", o.localConfigInfo.GetName())
```

Comment:
you need to change the test to `helper.CmdShouldFail` while checking for a closed session.

Reply:
Thanks!

Reply:
why are we not using log.Errorf() here?

Reply:
Because log.Errorf does not return, we need to return the error since we are using https://github.com/openshift/odo/blob/master/pkg/odo/util/cmdutils.go#L29 to properly exit, rather than relying on `log.Errorf()`

Reply:
OK got it.